tip-distributor/src/bin/claim-mev-tips.rs

```diff
@@ -117,61 +92,99 @@ async fn main() -> Result<(), ClaimMevError> {
+        }
+    }
+}
 
+#[tokio::main]
+async fn main() -> Result<(), ClaimMevError> {
+    env_logger::init();
+
+    gethostname()
+        .into_string()
+        .map(set_host_id)
+        .expect("set hostname");
+
+    let args: Args = Args::parse();
+    let keypair = Arc::new(read_keypair_file(&args.keypair_path).expect("read keypair file"));
+    let merkle_trees: GeneratedMerkleTreeCollection =
+        read_json_from_file(&args.merkle_trees_path).expect("read GeneratedMerkleTreeCollection");
+    let max_loop_duration = Duration::from_secs(args.max_retry_duration_secs);
+
+    info!(
+        "Starting to claim mev tips for epoch: {}",
+        merkle_trees.epoch
+    );
+    let epoch = merkle_trees.epoch;
+
+    let mut futs = vec![];
+    futs.push(tokio::spawn(start_mev_claim_process(
+        merkle_trees,
+        args.rpc_url.clone(),
+        args.tip_distribution_program_id,
+        keypair.clone(),
+        max_loop_duration,
+        args.micro_lamports,
+    )));
     if args.should_reclaim_rent {
-        let start = Instant::now();
-        match reclaim_rent(
-            args.rpc_url,
-            args.rpc_send_connection_count,
+        futs.push(tokio::spawn(start_rent_claim(
+            args.rpc_url.clone(),
             args.tip_distribution_program_id,
-            keypair,
-            args.max_loop_retries,
+            keypair.clone(),
             max_loop_duration,
             args.should_reclaim_tdas,
-            args.micro_lamports_per_compute_unit,
-        )
-        .await
-        {
-            Err(e) => {
-                datapoint_error!(
-                    "claim_mev_workflow-reclaim_rent_error",
-                    ("epoch", merkle_trees.epoch, i64),
-                    ("error", 1, i64),
-                    ("err_str", e.to_string(), String),
-                    (
-                        "merkle_trees_path",
-                        args.merkle_trees_path.to_string_lossy(),
-                        String
-                    ),
-                    ("elapsed_us", start.elapsed().as_micros(), i64),
-                );
-                Err(e)
-            }
-            Ok(()) => {
-                datapoint_info!(
-                    "claim_mev_workflow-reclaim_rent_completion",
-                    ("epoch", merkle_trees.epoch, i64),
-                    (
-                        "merkle_trees_path",
-                        args.merkle_trees_path.to_string_lossy(),
-                        String
-                    ),
-                    ("elapsed_us", start.elapsed().as_micros(), i64),
-                );
-                Ok(())
-            }
-        }?;
+            args.micro_lamports,
+            epoch,
+        )));
     }
+    let results = join_all(futs).await;
     solana_metrics::flush(); // sometimes last datapoint doesn't get emitted. this increases likelihood.
+    for r in results {
+        r.map_err(|e| ClaimMevError::UncaughtError { e: e.to_string() })??;
```

Comment:
could just unwrap these directly?

Reply:
main returns error, so just propagating that out